src/btree/bt_curnext.c

```diff
@@ -24,10 +24,10 @@ __cursor_fix_append_next(WT_CURSOR_BTREE *cbt, bool newpage, bool restart)
         goto restart_read;
 
     if (newpage) {
-        if ((cbt->ins = WT_SKIP_FIRST(cbt->ins_head)) == NULL)
+        if ((cbt->ins = __wt_skip_first(cbt->ins_head, WT_ATOMIC_ACQUIRE)) == NULL)
```

Comment:
I'm reading the standard (https://en.cppreference.com/w/cpp/atomic/memory_order) and trying to understand - I could use your help.
I *think* this is the same as using the `memory_order_acq_rel` - it's just specific because this is a load and not a store.

I wonder how we decide whether acquire release is adequate or if `seq_cst` is necessary. It seems as though `seq_cst` is necessary if some code related to this shared variable relies on consistent ordering across threads. i.e: using `ACQUIRE` here could violate an assumption in other code using `cbt->ins_head` that needs `seq_cst` - even though this code itself doesn't need the sequentially consistent ordering.

If that is right, doesn't it mean that *any* usage of a semantic weaker than `SEQ_CST` requires understanding fully all uses of the shared variable?

Reply:
```
I'm reading the standard (https://en.cppreference.com/w/cpp/atomic/memory_order) and trying to understand - I could use your help.
I think this is the same as using the memory_order_acq_rel - it's just specific because this is a load and not a store.
```
This is correct.

My understanding is that for the reader that doesn't require `seq_cst`, it can just use acquire and will not impact any other threads as it doesn't impact what data will other threads see (it is just reading). For the writer that may impact other threads that requires seq_cst, then itself needs to use seq_cst.